maven-artifact/pom.xml

```diff
@@ -45,10 +45,6 @@ under the License.
 
   <build>
     <plugins>
-      <plugin>
-        <groupId>org.codehaus.plexus</groupId>
-        <artifactId>plexus-component-metadata</artifactId>
-      </plugin>
```

Comment:
@josephw Any comments on this?

Reply:
This is the only cleanup from the more general work in #232 needed for this module. I've kept it in because I don't think it's large enough for its own issue and PR.

Reply:
I'd insist to move it to a separete, issueless, PR.

Reply:
Done; split into #258 (and this PR rebased).